logic/Rule.java

```diff
@@ -399,10 +399,14 @@ public Relation asRelation() {
                 SystemReference relationRef = SystemReference.of(0);
                 Identifier.Variable relationId = Identifier.Variable.of(relationRef);
                 traversal.isa(relationId, Identifier.Variable.label(relationType.getLabel().name()), false);
+                Set<Identifier.Variable> playersWithIds = new HashSet<>();
                 players.forEach(rp -> {
                     // note: NON-transitive role player types - we require an exact role being played
                     traversal.rolePlayer(relationId, rp.playerIdentifier, set(rp.roleType.getLabel()), rp.repetition);
-                    traversal.iid(rp.playerIdentifier, rp.player.getIID());
+                    if (!playersWithIds.contains(rp.playerIdentifier)) {
+                        traversal.iid(rp.playerIdentifier, rp.player.getIID());
+                        playersWithIds.add(rp.playerIdentifier);
+                    }
```

Comment:
this fixes a bug where if we had duplicate role player `(actor: $x, director: $x)`, we tried to assign an IID for `$x` twice, which is not allowed. we now we only add an ID for any role player once